src/components/FilterBar/FilterDisplay/Inputs/LazyMultiSelectInput.react.js

```diff
@@ -44,13 +44,30 @@ export class LazyMultiSelectInput extends React.Component {
         }
       },
       initSelection: function(element, callback) {
-        var data = [];
-        element.attr('value').split(',').forEach(function(value) {
-          data.push({id: value, text: value })
-        });
-        callback(data);
-      }
-    });
+        var values = [];
+        if(filter.itemUrl) {
+          Promise.all(element.attr('value').split(',').filter(Boolean).map(value => {
```

Comment:
If possible you can split the logic into a function name for easily read 
Ex: function handle_something